src/Auth.js

```diff
@@ -73,39 +72,33 @@ const renewSessionIfNeeded = async ({ config, session, sessionToken }) => {
   if (!config?.extendSessionOnUse) {
     return;
   }
-  clearTimeout(throttle[sessionToken]);
-  throttle[sessionToken] = setTimeout(async () => {
-    try {
-      if (!session) {
-        const { results } = await new RestQuery(
-          config,
-          master(config),
-          '_Session',
-          { sessionToken },
-          { limit: 1 }
-        ).execute();
-        session = results[0];
-      }
-      const lastUpdated = new Date(session?.updatedAt);
-      const yesterday = new Date();
-      yesterday.setDate(yesterday.getDate() - 1);
-      if (lastUpdated > yesterday || !session) {
-        return;
-      }
-      const expiresAt = config.generateSessionExpiresAt();
-      await new RestWrite(
-        config,
-        master(config),
-        '_Session',
-        { objectId: session.objectId },
-        { expiresAt: Parse._encode(expiresAt) }
-      ).execute();
-    } catch (e) {
-      if (e?.code !== Parse.Error.OBJECT_NOT_FOUND) {
-        logger.error('Could not update session expiry: ', e);
-      }
-    }
+  if (throttle[sessionToken]) {
+    return;
+  }
+  throttle[sessionToken] = true;
+  setTimeout(() => {
+    delete throttle[sessionToken];
```

Comment:
we still have a memory issue in case of huge amount of different tokens. For this kind of use case a LRU is a good idea to define a max store size.

Also do we really need a time out . Non tracked timeout ( causing a desync in request/response) can introduce issues with test systems ( since timeouts are not properly cleared).

The best solution here is to remove the timeout, and just use a standard LRU cache , with a max size and a max entry validity type of 500ms if you want a cache with a validity of 500ms.

Custom store combined with timemout or intervals are generally a bad idea.